nano/node/active_transactions.cpp

```diff
@@ -579,6 +589,71 @@ void nano::active_transactions::erase (nano::block const & block_a)
 	}
 }
 
+bool nano::active_transactions::should_flush ()
+{
+	bool result (false);
+	counter.trend_sample ();
+	size_t minimum_size (1);
+	std::unique_lock<std::mutex> counter_lock (counter.mutex);
+	if (counter.rate == 0)
+	{
+		//set minimum size to 4 for test network
+		minimum_size = node.network_params.network.is_test_network () ? 4 : 512;
+	}
+	else
+	{
+		minimum_size = counter.rate * 512;
+	}
+	if (roots.size () > minimum_size)
```

Comment:
Not sure if there can be some upper limit as well, i.e. 100k or something

Reply:
added early return if over 100k in roots